crates/uv-auth/src/lib.rs

```diff
@@ -35,3 +35,45 @@ pub fn store_credentials_from_url(url: &Url) -> bool {
         false
     }
 }
+
+/// Populate the global authentication store with credentials from the environment, if any.
+///
+/// Respects `UV_AUTH_URLS` with whitespace separated URLs.
+pub fn store_credentials_from_environment() -> bool {
+    let Some(urls) = std::env::var_os("UV_AUTH_URLS") else {
+        return false;
+    };
+
+    let mut populated = false;
+    for mut url in urls
+        .to_string_lossy()
+        .split_ascii_whitespace()
+        .filter_map(parse_url_from_env)
+    {
+        if let Some(credentials) = Credentials::from_url(&url) {
+            CREDENTIALS_CACHE.insert_url(&url, Arc::new(credentials));
+            // Redact the password for display
+            if url.password().is_some() {
+                let _ = url.set_password(Some("***"));
+            }
+            debug!("Added credentials for `{url}`");
+            populated = true;
+        } else {
+            debug!("Ignoring URL without credentials in `UV_AUTH_URLS`: {url}");
+        }
+    }
+
+    populated
+}
+
+/// Parse a URL, allowing the scheme to be missing (and inferred as `https://`)
+fn parse_url_from_env(url: &str) -> Option<Url> {
+    if url.starts_with("https://") {
+        Url::parse(url)
+    } else {
+        let url = format!("https://{url}");
+        Url::parse(&url)
+    }
+    .inspect_err(|err| debug!("Ignoring invalid URL in `UV_AUTH_URLS`: {err}"))
+    .ok()
+}
```

Comment:
@BurntSushi recommendations here?

Reply:
(We can't try parsing and reparse on a missing scheme error because `public:test@foo.com` treats `public` as a scheme.)

Reply:
What's wrong with what's here?

Reply:
Uhh, well if you do `http://foo` we don't want to support that but we'll fail by doing `https://http://foo` which seems weird. I guess that's fine though?

Reply:
I'll just special case http for now.

Reply:
Yeah I don't have any better ideas here, especially if URL parsing is treating things as schemes that we don't want to be treated as schemes. So I'd guess you'd probably need to write a custom URL parser to deal with this in a way that is completely satisfying. Special casing `http` seems fine.